dogpile/cache/backends/memcached.py

```diff
@@ -509,15 +555,32 @@ def __init__(self, arguments):
         self.serde = arguments.get("serde", pymemcache.serde.pickle_serde)
         self.default_noreply = arguments.get("default_noreply", False)
         self.tls_context = arguments.get("tls_context", None)
+        self.socket_keepalive = arguments.get("socket_keepalive", None)
+        self.retry_client = arguments.get("retry_client", False)
+        self.retry_attempts = arguments.get("retry_attempts", 2)
+        self.retry_delay = arguments.get("retry_delay", 0)
+        self.retry_for = arguments.get("retry_for", None)
+        self.do_not_retry_for = arguments.get("do_not_retry_for", None)
 
     def _imports(self):
         global pymemcache
         import pymemcache
 
     def _create_client(self):
-        return pymemcache.client.hash.HashClient(
+        client = pymemcache.client.hash.HashClient(
             self.url,
             serde=self.serde,
             default_noreply=self.default_noreply,
             tls_context=self.tls_context,
+            socket_keepalive=self.socket_keepalive,
         )
+        if self.retry_client:
```

Comment:
do we want to raise an argument error if they passed any of the "Retry" options without passing "retry_client" ?

Reply:
Yes from a UX point of view that's make sense.